src/groupeddataframe/splitapplycombine.jl

```diff
@@ -591,23 +620,25 @@ function _combine(gd::GroupedDataFrame,
     # here first field in trans_res[i] is used to keep track how the column was generated
     # a correct index is stored in idx variable
 
-    for i in eachindex(trans_res)
-        col_idx = trans_res[i].col_idx
-        col = trans_res[i].col
-        if keeprows && col_idx !== idx_keeprows # we need to reorder the column
-            newcol = similar(col)
-            # we can probably make it more efficient, but I leave it as an optimization for the future
-            gd_idx = gd.idx
-            k = 0
-            # consider adding @inbounds later
-            for (s, e) in zip(gd.starts, gd.ends)
-                for j in s:e
-                    k += 1
-                    newcol[gd_idx[j]] = col[k]
+    @sync for i in eachindex(trans_res)
```

Comment:
can you please add a comment on the logic how we handle the threading here? (i.e. a high-level logic how we ensure that we do not have race-conditions and correctly process things). Thank you!

Reply:
I'm not sure what to say about this loop actually. Each entry can be processed in parallel because we extract its fields, do some computing based on them, and replace the original entry with the result. We don't update any external state. Is there something in particular that you feel is worth mentioning?

Reply:
I do not mean this loop but the logic in general. I.e. that we have to:
1. sync `idx_agg` across threads
2. do actual inserting of columns in post-processing as we have to do it sequentially (although we compute them in parallel)

maybe something more that might be considered hard to understand.